src/mono/wasi/runtime/CMakeLists.txt

```diff
@@ -14,8 +14,10 @@ set_target_properties(dotnet PROPERTIES COMPILE_FLAGS ${CONFIGURATION_WASICC_FLA
 
 target_link_libraries(dotnet
     PRIVATE
-    ${CMAKE_SYSROOT}/lib/wasm32-wasi/libc++.a
-    ${CMAKE_SYSROOT}/lib/wasm32-wasi/libc++abi.a
+    ${CMAKE_SYSROOT}/lib/wasm32-wasip2/libc++.a
+    ${CMAKE_SYSROOT}/lib/wasm32-wasip2/libc++abi.a
+    # TODO this makes hard dependency on wasi:http@0.2.0, ideally this would be trimmable when HTTP is not used
```

Comment:
This is unfortunate, it will require HTTP to be always enabled on the host for Mono.
I think we can make it optional by re-linking the final app.

Reply:
Also the binary `.o` is not ideal

Reply:
I opened an issue about the component type object file: https://github.com/bytecodealliance/wit-bindgen/issues/994

Reply:
@lewing @akoeplinger do you know if this will cause mayhem if we merge the binary file ? 
Some Azdo check or something ?
Or we need to block this until we have source-build ?

Reply:
Yes, checked in binary file like this will fail multiple gates. It will fail a gate in source build in dotnet/sdk. I would expect it to fail some internal AzDo gates that check for checked binaries with unknown origin too.

Reply:
I've started working on https://github.com/bytecodealliance/wit-bindgen/issues/994 and should have something usable by the time @pavelsavara returns from his time off.

Reply:
@lewing suggested that we should be generating the imports from `[DLLImport]` as C file
https://github.com/dotnet/runtime/blob/bf54567d4ed4d18c4a99059ba8c948515754143f/src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs#L284-L288

I'm looking at why is that not working

Reply:
I now use `.wit` world instead of `.o` binary. I rebased this branch to make sure that the binary is not in the commit history.